packages/db/src/translate/granules.ts

```diff
@@ -116,18 +118,45 @@ export const translatePostgresGranuleToApiGranule = async ({
   return apiGranule;
 };
 
+const returnNullOrUndefinedOrDate = (dateVal: string | number | null | undefined) => {
+  if (dateVal === undefined) return undefined;
+  if (dateVal === null) return dateVal;
+  return new Date(dateVal);
```

Comment:
How about: 1babc4c61a

We can't just reduce that entirely to a ternary as we *want* to return null and a ternary evaluates dateVal as truthy/falsy.

Reply:
`return dateVal ? new Date(dateVal) : dateVal; ` should cover both `null` and `undefined`

Reply:
@jennyhliu  The intent is that we don't *want* to cover null here.

Reply:
Unfortunately it's not explicit enough for the typing and probably incorrect -- returning explicit values over implicit is *always* recommended, and in this case required.   Typescript interprets the ternary return without an explicit logic gate as  string | number | Date | undefined, because `''` and `0` aren't correctly evaluated by a ternary. 

e.g.: 

```
> 0 ? 'foo' : 'bar'
'bar'
> 1 ? 'foo' : 'bar'
'foo'
> '' ? 'foo' : 'bar'
'bar'
> 'someString' ? 'foo' : 'bar'
'foo'
```

Using ternary here  doesn't ensure explicit typing that the compiler can understand, as the return values aren't ensured.   I can use the ternary, but then I need to explicitly evaluate the return in the translation method, which is *far* less readable.  

This sort of catch is one of the things that makes Typescript awesome 🎉 .     More generally the fix for this line of thought would be nullish coalescing, but that can't be negated and in this instance we *want* null/undefined....so: 

3ed988fa1f